src/validation.cpp

```diff
@@ -3300,8 +3311,6 @@ bool CChainState::AcceptBlockHeader(const CBlockHeader& block, CValidationState&
     if (ppindex)
         *ppindex = pindex;
 
-    CheckBlockIndex(chainparams.GetConsensus());
```

Comment:
Note to reviewers: this is moved to the two callsites below to avoid making reference to a specific CChainState in BlockManager.

Reply:
If this refactoring seems fragile in light of silent merge conflicts, an alternative would be to pass in a reference to the chainstate and then call CBI on that reference. This way, a new call to `AcceptBlockHeader` introduced before your change will not be a silent merge conflict if merged with your chagnes.

Reply:
As @sdaftuar pointed out in https://github.com/bitcoin/bitcoin/issues/16444#issuecomment-514801708 `AcceptBlockHeader` can return early, so we're now calling `CheckBlockIndex(()` more often than before.